tensorboard/webapp/plugins/npmi/views/data_selection/metric_arithmetic/metric_arithmetic_element/metric_arithmetic_element_container.ts

```diff
@@ -0,0 +1,102 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Component, ChangeDetectionStrategy, Input, OnInit} from '@angular/core';
+import {select, Store} from '@ngrx/store';
+
+import {map} from 'rxjs/operators';
+
+import {State} from '../../../../store/npmi_types';
+import {getMetricFilters} from '../../../../store';
+import * as npmiActions from '../../../../actions';
+
+/** @typehack */ import * as _typeHackRxjs from 'rxjs';
+
+@Component({
+  selector: 'npmi-metric-arithmetic-element',
+  template: `
+    <metric-arithmetic-element-component
+      [metric]="metric"
+      [filterValues]="filterValues"
+      [minFilterValid]="minFilterValid"
+      [maxFilterValid]="maxFilterValid"
+      (onRemove)="remove($event)"
+      (onFilterChange)="filterChange($event)"
+    ></metric-arithmetic-element-component>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MetricArithmeticElementContainer {
+  @Input() metric!: string;
+  filterValues = {min: '-1.0', max: '1.0'};
+  readonly filterValueSetter$ = this.store.pipe(select(getMetricFilters)).pipe(
+    map((filters) => {
+      const filter = filters[this.metric];
+      const min = filter.includeNaN
+        ? 'NaN'
+        : this.rounded(filter.min).toString();
+      const max =
+        filter.max < filter.min ? 'NaN' : this.rounded(filter.max).toString();
+      this.filterValues = {min: min, max: max};
+    })
+  );
+  minFilterValid = true;
+  maxFilterValid = true;
+
+  constructor(private readonly store: Store<State>) {}
+
+  remove(metric: string) {
+    this.store.dispatch(npmiActions.npmiRemoveMetricFilter({metric: metric}));
+  }
+
+  filterChange(newValues: {min: string; max: string}) {
+    this.filterValues = {min: newValues.min, max: newValues.max};
+    this.minFilterValid = this.entryValid(newValues.min);
+    this.maxFilterValid =
+      this.entryValid(newValues.max) &&
+      (parseFloat(newValues.max) >= parseFloat(newValues.min) ||
+        newValues.min == 'NaN');
+    if (this.minFilterValid && this.maxFilterValid) {
+      let min = newValues.min === 'NaN' ? -1 : parseFloat(newValues.min);
+      let max = newValues.max === 'NaN' ? -2 : parseFloat(newValues.max);
+      let includeNaN = newValues.min === 'NaN';
+      this.store.dispatch(
+        npmiActions.npmiChangeMetricFilter({
+          metric: this.metric,
+          max: max,
+          min: min,
+          includeNaN: includeNaN,
+        })
+      );
+    }
+  }
+
+  private entryValid(value: string) {
+    if (value === 'NaN') {
+      return true;
+    } else {
+      let numberValue = parseFloat(value);
+      if (numberValue === NaN) {
+        return false;
+      } else if (numberValue >= -1.0 && numberValue <= 1.0) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private rounded(value: number): number {
```

Comment:
rename this to `roundToThreeDecimalPoints`?

Reply:
Done.